runtime/vm/CRIUHelpers.cpp

```diff
@@ -254,24 +255,36 @@ static void
 initializeCriuHooks(J9VMThread *currentThread)
 {
 	J9JavaVM *vm = currentThread->javaVM;
+	J9Class *juRandomClass = NULL;
 
 	Trc_VM_criu_initHooks_Entry(currentThread);
 	cleanupCriuHooks(currentThread);
 	if (NULL == vm->checkpointState.hookRecords) {
 		vm->checkpointState.hookRecords = pool_new(sizeof(J9InternalHookRecord), 0, 0, 0, J9_GET_CALLSITE(), OMRMEM_CATEGORY_VM, POOL_FOR_PORT(vm->portLibrary));
 		if (NULL == vm->checkpointState.hookRecords) {
 			setNativeOutOfMemoryError(currentThread, 0, 0);
+			goto done;
 		}
 	}
-	if (NULL != vm->checkpointState.hookRecords) {
-		/* Add restore hook to re-seed java.uti.Random.seed.value. */
+
+	if (NULL == vm->checkpointState.delayedLockingOperationsRecords) {
+		vm->checkpointState.delayedLockingOperationsRecords = pool_new(sizeof(J9DelayedLockingOpertionsRecord), 0, 0, 0, J9_GET_CALLSITE(), OMRMEM_CATEGORY_VM, POOL_FOR_PORT(vm->portLibrary));
+		if (NULL == vm->checkpointState.delayedLockingOperationsRecords) {
+			setNativeOutOfMemoryError(currentThread, 0, 0);
+			goto done;
+		}
+	}
+
+
+	/* Add restore hook to re-seed java.uti.Random.seed.value */
 #define JAVA_UTIL_RANDOM "java/util/Random"
-		J9Class *juRandomClass = hashClassTableAt(vm->systemClassLoader, (U_8 *)JAVA_UTIL_RANDOM, LITERAL_STRLEN(JAVA_UTIL_RANDOM));
+	juRandomClass = peekClassHashTable(currentThread, vm->systemClassLoader, (U_8 *)JAVA_UTIL_RANDOM, LITERAL_STRLEN(JAVA_UTIL_RANDOM));
```

Comment:
```suggestion
	J9Class *juRandomClass = peekClassHashTable(currentThread, vm->systemClassLoader, (U_8 *)JAVA_UTIL_RANDOM, LITERAL_STRLEN(JAVA_UTIL_RANDOM));
```

Reply:
I cant move this one because of the jump to `done:`

Reply:
It might require an additional scope so the change becomes:
```c
{
	J9Class *juRandomClass = peekClassHashTable(currentThread, vm->systemClassLoader, (U_8 *)JAVA_UTIL_RANDOM, LITERAL_STRLEN(JAVA_UTIL_RANDOM));
....
	if (NULL != juRandomClass) {
		addInternalJVMCheckpointHook(currentThread, TRUE, juRandomClass, FALSE, juRandomReseed);
	}
}

Reply:
But I'm OK if you want to leave it as is.  The method's clear enough that this isn't critical